debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresChangeRecordEmitter.java

```diff
@@ -54,6 +61,7 @@ public PostgresChangeRecordEmitter(OffsetContext offset, Clock clock, PostgresCo
         this.connection = connection;
 
         this.tableId = PostgresSchema.parse(message.getTable());
+        this.isJsonPlugin = "wal2json".equals(connectorConfig.plugin().getPostgresPluginName());
```

Comment:
Instead of checking the plug-in name, I'd rather suggest adding a method to `LogicalDecoder` that describes what you're actually after, e.g. `conveysUnchangedToastColumnValueMarker()` which would return true for pgoutput and decoderbufs and false for wal2json.

Reply:
@gunnarmorling of course, sorry for that

Reply:
@gunnarmorling Changed